src/stream/stream/mod.rs

```diff
@@ -313,6 +315,41 @@ extension_trait! {
             TakeWhile::new(self, predicate)
         }
 
+        #[doc = r#"
+            Limit the amount of items yielded per timeslice in a stream.
+
+            # Examples
+            ```ignore
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::stream;
+            use std::time::Duration;
+
+            // emit value every 1 second
+            let s = stream::interval(Duration::from_nanos(1000000)).enumerate();
+
+            // throttle for 2 seconds
+            let s = s.throttle(Duration::from_secs(2));
+
+            s.for_each(|(n, _)| {
+                dbg!(n);
+            })
```

Comment:
We could use `.fold` instead of `for_each` here to gather the total time waited into a single duration. Then after we're done running we can verify that the total time elapsed was within a certain bound.